api/src/repositories/submission-repository.ts

```diff
@@ -1731,4 +1733,109 @@ export class SubmissionRepository extends BaseRepository {
 
     return response.rows[0].value;
   }
+
+  /**
+   * Build the base query for submission features.
+   *
+   * @param {number} submissionId
+   * @param {Knex} knex
+   * @return {Knex.QueryBuilder}
+   * @memberof SubmissionRepository
+   */
+  private _getSubmissionFeaturesBaseQuery(submissionId: number, knex: Knex) {
+    const baseQuery = knex('submission_feature')
+      .select(
+        'submission_feature.submission_id',
+        'submission_feature.submission_feature_id',
+        'submission_feature.feature_type_id',
+        knex.raw('feature_type.name AS feature_type_name'),
+        knex.raw(`
+        EXISTS (
+          SELECT 1 
+          FROM submission_feature_security sfs
+          WHERE sfs.submission_feature_id = submission_feature.submission_feature_id
+        ) AS secured
+      `)
+      )
+      .leftJoin('feature_type', 'feature_type.feature_type_id', 'submission_feature.feature_type_id')
+      .where('submission_feature.submission_id', submissionId);
+
+    return knex.with('base_features', baseQuery).distinct('*').from('base_features').orderBy('submission_feature_id');
```

Comment:
[SHOULD FIX] Base query has hardcoded `.orderBy('submission_feature_id')`. When `applyPagination` adds another `.orderBy()`, Knex appends (doesn't replace), so final SQL is:
```sql
ORDER BY submission_feature_id, feature_type_name desc
```
Primary sort is always `submission_feature_id`, making user-selected sort ineffective.

**Fix:** Remove hardcoded `orderBy` from base query, let `applyPagination` handle it:
```suggestion
    return knex.with('base_features', baseQuery).distinct('*').from('base_features');
```